test/src/test/kotlin/graphql/nadel/tests/hooks/chain-input-rename-with-argument-transform.kt

```diff
@@ -0,0 +1,93 @@
+import graphql.nadel.engine.transform.result.NadelResultInstruction
+import graphql.nadel.engine.transform.result.json.JsonNodes
+import graphql.nadel.engine.util.toBuilder
+import graphql.nadel.tests.EngineTestHook
+import graphql.nadel.tests.UseHook
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.NormalizedInputValue
+
+@UseHook
+class `chain-input-rename-with-object-argument-transform` : EngineTestHook {
+    override val customTransforms: List<NadelTransform<out Any>>
+        get() = listOf(
+            object : NadelTransform<Unit> {
+                override suspend fun isApplicable(
+                    executionContext: NadelExecutionContext,
+                    executionBlueprint: NadelOverallExecutionBlueprint,
+                    services: Map<String, Service>,
+                    service: Service,
+                    overallField: ExecutableNormalizedField,
+                    hydrationDetails: ServiceExecutionHydrationDetails?,
+                ): Unit? {
+                    return Unit.takeIf {
+                        overallField.normalizedArguments.isNotEmpty()
+                    }
+                }
+
+                override suspend fun transformField(
+                    executionContext: NadelExecutionContext,
+                    transformer: NadelQueryTransformer,
+                    executionBlueprint: NadelOverallExecutionBlueprint,
+                    service: Service,
+                    field: ExecutableNormalizedField,
+                    state: Unit,
+                ): NadelTransformFieldResult {
+                    return NadelTransformFieldResult(
+                        newField = field.toBuilder()
+                            .normalizedArguments(
+                                field.normalizedArguments
+                                    .mapValues { (_, value) ->
+                                        require(value.typeName == "StartSprintInput")
+
+                                        NormalizedInputValue(
+                                            "StartSprintInput",
+                                            mapOf(
+                                                "boardId" to NormalizedInputValue("ID!", StringValue("123")),
+                                                "sprintId" to NormalizedInputValue("ID!", StringValue("456")),
+                                                "name" to NormalizedInputValue("String!", StringValue("Test Input")),
+                                                "goal" to NormalizedInputValue("String",
+                                                    NullValue.newNullValue().build()),
+                                                "startDate" to NormalizedInputValue("String!",
+                                                    StringValue("2022-03-22")),
+                                                "endDate" to NormalizedInputValue("String!", StringValue("2022-04-02")),
```

Comment:
Why doo you need to hard code a rename here?  Can we not specify that in tests?

As a reader of this class - I cant tell WHY we have a complex test transformation on place

I think a comment on WHY this test code exists will help future generations

Reply:
Yup fair will add a comment. The transform is supposed to simulate what the ARI transform does.